.github/workflows/demo.yml

```diff
@@ -0,0 +1,60 @@
+name: test notebooks
+
+# this aims to test the notebooks, the way the user would
+# - install package from github
+# - use the nbmake pytest plugin to run the notebooks from top to bottom
+
+#
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: ["*"]
+
+jobs:
+
+  test_notebooks:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macOS-10.15, windows-latest]
+        python-version: ["3.9"]
```

Comment:
note that this only runs the demo on python 3.9.

I could switch to the latest python if #18 is solved